libs/component-library/src/lib/tabs/tabs.tsx

```diff
@@ -67,19 +120,56 @@ type FxTabsProps = {
   onSelect: (idx: number) => void;
   animate?: boolean;
   animationDuration?: number; // ms
+  variant?: 'fixed' | 'auto';
 };
 export const FxTabs = ({
   items,
   selectedIdx = 0,
   onSelect,
   animate = true,
   animationDuration = ANIMATION_DURATION,
+  variant = 'fixed',
 }: FxTabsProps) => {
   const translateX = useSharedValue(0);
-  const containerWidth = useSharedValue<number>(0);
-  const highlightWidth = useDerivedValue(() => {
-    return containerWidth.value / items.length;
-  }, [containerWidth, items]);
+  const highlightWidth = useSharedValue<number>(0);
+  const selectedLayout = useSharedValue<{ x: number; width: number }>({
+    x: 0,
+    width: 0,
+  });
+  const itemLayouts = React.useRef<Array<{ x: number; width: number }>>([]);
+
+  React.useEffect(() => {
+    if (itemLayouts.current[selectedIdx]) {
+      selectedLayout.value = itemLayouts.current[selectedIdx];
+    }
+  }, [selectedIdx]); // eslint-disable-line react-hooks/exhaustive-deps
+
+  const tabsItems = items.map((item, idx) => {
+    return (
+      <View
+        key={idx}
+        style={styles[variant]}
```

Comment:
Along my suggestion with the updated tabsItems, this `View` could be `FxBox` and ditch the `styleSheet` `styles`

```diff
-      <View
-        key={idx}
-        style={styles[variant]}
+        <FxBox
+          key={idx}
+          flex={isAuto ? undefined : 1}
```
.
.
.

```diff
-const styles = StyleSheet.create({
-  fixed: {
-    flex: 1,
-  },
-  auto: {},
-});
```